src/lib.rs

```diff
@@ -393,8 +401,73 @@ impl<L: SynthLanguage> Synthesizer<L> {
 
         // initialize egraph with variables and constants
         L::init_synth(&mut synth);
-        synth.initial_egraph = synth.egraph.clone();
 
+        if num_ces > 0 {
```

Comment:
you can just do `params.num_ces` here, I don't think you need to make a local variable for it

Reply:
weird rust ownership stuff, may be a more elegant way to do this